src/robot/libraries/Collections.py

```diff
@@ -27,18 +32,33 @@
 
 NOT_SET = NotSet()
 
+I_ = TypeVar("I_", bound=Iterable)
+M_ = TypeVar("M_", bound=Mapping)
+IC = Union[
+    bool, Literal["key", "KEY", "keys", "KEYS", "value", "VALUE", "values", "VALUES"]
+]
+
 
 class _List:
 
-    def convert_to_list(self, item):
+    # NOTE: Annotation with list would cause the keyword to fail on string arguments.
+    # This would break current behavior. Annotation with "list | str" would however
+    # cause weird behvior with non-iterable arguments since their string representation
+    # would be converted to a list and returned. For this reason, object is used as an
+    # annotation, effectively preventing automatic argument conversion.
+    # In the future support for string arugments will be dropped;
+    # 'Split String To Characters' from the String library can be used instead.
+    # This change in behavior does however require a deprecation period.
+    def convert_to_list(self, item: object) -> list:
         """Converts the given ``item`` to a Python ``list`` type.
 
         Mainly useful for converting tuples and other iterable to lists.
         Use `Create List` from the BuiltIn library for constructing new lists.
         """
-        return list(item)
+        return list(item)  # type: ignore
 
-    def append_to_list(self, list_, *values):
+    # NOTE: https://github.com/robotframework/robotframework/issues/5536
```

Comment:
We have the issue so this comment is unnecessary. The plan is to implement the issue as part of RF 7.4 and the comment can be removed at that point, but I still think the comment is unnecessary.